### PR TITLE
Add crowdsource

### PR DESCRIPTION
**Pull request by milanocookies93**

### CODE REVIEW

**Comment by hsingh23** [comment]
Did you mean to change the password?

**Comment by milanocookies93** [comment]
I don't want to put the password on github, the text there is a placeholder. On the machine that chroma runs on I will put the real password in that place.

**Comment by hsingh23** [comment]
There is a lot of code repetition. Consider refactoring in your own time/

File: animations/crowdsource/main.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+black,black,black,black,black,black,
+black,black,black,black,black,black,
+black,black,black,black,black,black,
+black,black,black,black,black,black,
+black,black,black,black,black,black,
+black,black,black,black,black,black,
+black,black,black,black,black,black]
+
+if __name__ == "__main__":
+    password = "PASSWORD GOES HERE"
```
Comment (bnookala) [inline animations/crowdsource/main.py]: you can put the password in a file, identified by an environment variable. this way, you're not modifying scripts on a production machine.

File: animations/crowdsource/main.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+black = (0.0,0.0,0.0)
+green = (0.0, 1023.0, 0.0)
+red = (1023.0, 0.0, 0.0)
+blue = (0.0, 0.0, 1023.0)
+orange = (1023.0, 511.5,0.0)
+yellow = (1023.0, 1023.0, 0.0)
+purple = (1023.0, 0.0, 1023.0)
+teal = (511.5, 1023.0, 1023.0)
+
+pixels = [black,black,black,black,black,black,
```
Comment (bnookala) [inline animations/crowdsource/main.py]: pixels = [black] * 48

File: animations/crowdsource/main.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                    out.write(pixels)
+                if subject == "yellow":
+                    time.sleep(1.0)
+                    pixels[0] = yellow
+                    out.write(pixels)
+                    time.sleep(1.0)
+                    random.shuffle(pixels)
+                    pixels[0] = black
+                    out.write(pixels)
+                if subject == "purple":
```
Comment (bnookala) [inline animations/crowdsource/main.py]: the colors should probably be coded as an enum object, ie:

    Colors(object):
        PURPLE = "purple"
        YELLOW = "yellow"

etc.

File: animations/crowdsource/main.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            for num in data[0].split():
+                rv, data = M.fetch(num, '(RFC822)')
+                if rv != 'OK':
+                    M.close()
+                    sys.exit(1)
+        
+                msg = email.message_from_string(data[0][1])
+                decode = email.header.decode_header(msg['Subject'])[0]
+                subject = unicode(decode[0])
+                if subject == "red":
```
Comment (bnookala) [inline animations/crowdsource/main.py]: the content of each of these if blocks is very redundant

    time.sleep(1.0)
    pixels[0] = SOME_COLOR
    out.write(pixels)
    time.sleep(1.0)
    random.shuffle(pixels)
    pixels[0] = black
    out.write(pixels)

why not just do:

    time.sleep(1.0)
    if subject == Colors.Black:
        pixels[0] = black
    elif subject == Colors.Blue:
        pixels[0] = blue
    …

    out.write(pixels)
    time.sleep(1.0)
    random.shuffle(pixels)
    pixels[0] = black
    out.write(pixels)

File: animations/crowdsource/manifest.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+	"name":"Crowdsource",
```
Comment (bnookala) [inline animations/crowdsource/manifest.json]: space between key and value:

"name": "Crowdsource",
etc.

**Comment by milanocookies93** [comment]
@bnookala, thanks for the suggestions. Some of these I've already dealt with in the next PR that was merged, and some I'm going to be dealing with when I get some more free time.

**Pull request merged**